feature/select-subject-age-group/src/main/java/com/simprints/feature/selectagegroup/screen/SelectSubjectAgeGroupFragment.kt

```diff
@@ -0,0 +1,66 @@
+package com.simprints.feature.selectagegroup.screen
+
+import android.os.Bundle
+import android.view.View
+import androidx.activity.addCallback
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.simprints.feature.exitform.toArgs
+import com.simprints.feature.selectagegroup.R
+import com.simprints.feature.selectagegroup.SelectSubjectAgeGroupResult
+import com.simprints.feature.selectagegroup.databinding.FragmentAgeGroupSelectionBinding
+import com.simprints.infra.config.store.models.AgeGroup
+import com.simprints.infra.uibase.navigation.finishWithResult
+import com.simprints.infra.uibase.navigation.navigateSafely
+import com.simprints.infra.uibase.viewbinding.viewBinding
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+internal class SelectSubjectAgeGroupFragment : Fragment(R.layout.fragment_age_group_selection) {
+
+    private val viewModel: SelectSubjectAgeGroupViewModel by viewModels()
+    private val binding by viewBinding(FragmentAgeGroupSelectionBinding::bind)
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.ageGroups.observe(viewLifecycleOwner) { ageGroupsList ->
+            fillRecyclerView(ageGroupsList)
+        }
+
+        viewModel.showExitForm.observe(viewLifecycleOwner) { exitFormConfig ->
```

Comment:
Could be using the custom observer instance to avoid checking "ifNotHandled" manually.

Reply:
I just copied the code from the consent fragment

Reply:
Then consent fragment could also be improved :)